contracts/eternal-storage/OwnedEternalStorage.sol

```diff
@@ -0,0 +1,65 @@
+pragma solidity 0.5.2;
+
+import "./EternalStorage.sol";
+
+
+/// @title Owned Contracts
+/// @dev This contract provides ownership and access control functionality.
+/// Using the `onlyOwner` modifier, a function can be restricted to being
+/// called by the owner of the contract.  The owner of a contract can
+/// irrevocably transfer ownership using the `transferOwnership` function.
+contract Owned {
```

Comment:
Let's delete the `Owned` contract and leave only `OwnedEternalStorage` which will contain only the `onlyOwner` modifier and storage mappings (with `_version`, `_implementation`, and `_owner` private members). Other functions (such as `transferOwnership`, `getOwner`, and `OwnershipTransferred`) must be left inside the `EternalStorageProxy` contract.

Reply:
We shouldn't include `transferOwnership`, `getOwner`, and `OwnershipTransferred` to `OwnedEternalStorage` because these functions must only be used inside the `EternalStorageProxy` (not inside implementation). Otherwise we get increased size of implementation's code and also get clashing: https://medium.com/nomic-labs-blog/malicious-backdoors-in-ethereum-proxies-62629adf3357

Reply:
> We shouldn't include `transferOwnership`, `getOwner`, and `OwnershipTransferred` to `OwnedEternalStorage` because these functions must only be used inside the `EternalStorageProxy` (not inside implementation). Otherwise we get increased size of implementation's code and also get clashing: https://medium.com/nomic-labs-blog/malicious-backdoors-in-ethereum-proxies-62629adf3357

Clashing is not actually an issue here, as the functions will be called on the proxies (which is what we want).  That said, it makes auditing more difficult, and it does also increase the code size.

Reply:
Of course, I meant `internal`.

Reply:
> May I ask why?

Because those functions shouldn't be in the implementations. It's enough to have them in EternalStorageProxy. The `EternalStorageProxy` inherits `OwnedEternalStorage` in the same way as implementation, so we get duplicated code which shouldn't be in implementation.

Reply:
@varasev could we include unit-tests that check that every proxied function can, in fact, be called?